File: NopTHAndroid/Cau3_MySelf2/app/src/main/java/ntu/thu_62132086/cau3_myself/MainActivity.java
package ntu.thu_62132086.cau3_myself;


import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Initialize TextViews
        TextView textViewHometown = findViewById(R.id.textViewHometown);
        TextView textViewDOB = findViewById(R.id.textViewDOB);
        TextView textViewHobbies = findViewById(R.id.textViewHobbies);

        // Set student information
        textViewHometown.setText("Quê quán: (Điền thông tin quê quán)");
        textViewDOB.setText("Ngày sinh: (Điền thông tin ngày sinh)");
        textViewHobbies.setText("Sở thích: (Liệt kê các sở thích của bạn)");
    }
}

File: NopTHAndroid/DrawerNavigationWithFragment/app/src/main/java/ntu/thu_62132086/drawernavigationwithfragment/MainActivity.kt
package ntu.thu_62132086.drawernavigationwithfragment

import android.os.Bundle
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import com.google.android.material.navigation.NavigationView


class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
    private var drawerLayout: DrawerLayout? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        toolbar.setBackgroundColor(getColor(R.color.lavender))
        setSupportActionBar(toolbar)
        drawerLayout = findViewById<DrawerLayout>(R.id.drawer_layout)
        val navigationView = findViewById<NavigationView>(R.id.nav_view)
        navigationView.setNavigationItemSelectedListener(this)
        val toggle = ActionBarDrawerToggle(
            this,
            drawerLayout,
            toolbar,
            R.string.open_nav,
            R.string.close_nav
        )
        drawerLayout.addDrawerListener(toggle)
        toggle.syncState()
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, HomeFragment()).commit()
            navigationView.setCheckedItem(R.id.nav_home)
        }
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        if (id == R.id.nav_home) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, HomeFragment()).commit()
        } else if (id == R.id.nav_about) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, AboutFragment()).commit()
        } else if (id == R.id.nav_settings) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, SettingsFragment()).commit()
        } else if (id == R.id.nav_share) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, ShareFragment()).commit()
        } else if (id == R.id.nav_about) {
            Toast.makeText(this, "Logout!", Toast.LENGTH_SHORT).show()
        }
        drawerLayout!!.closeDrawer(GravityCompat.START)
        return true
    }

    override fun onBackPressed() {
        if (drawerLayout!!.isDrawerOpen(GravityCompat.START)) {
            drawerLayout!!.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }
}